lib/auth/register.go

```diff
@@ -91,6 +92,10 @@ type RegisterParams struct {
 	CAPath string
 	// GetHostCredentials is a client that can fetch host credentials.
 	GetHostCredentials HostCredentials
+	// Time specifies the time as number of seconds since epoch to use in
+	// client for verifying TLS certificates.
+	// If unspecified, time.Now() will be used
+	Time func() time.Time
```

Comment:
Why not pass a `clockwork.Clock`?

Reply:
There was no immediate need for the whole clock - just the function that returns time. This is supposed to mirror the [tls.Config.Time](https://github.com/golang/go/blob/9b955d2d3fcff6a5bc8bce7bafdc4c634a28e95b/src/crypto/tls/common.go#L491-L493).